File: yunyi/209.长度最小的子数组.cpp
/*
 * @lc app=leetcode.cn id=209 lang=cpp
 *
 * [209] 长度最小的子数组
 */

// @lc code=start
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int len = 1;
        int start = 0;
        int sum = 0;
        while (len <= nums.size()) {
            start = 0;
            sum = 0;
            while (start <= nums.size()-len) {
                for (int i=start;i<start+len;i++) {
                    sum += nums[i];
                }
                if (sum >= target) return len;
                else {
                    start += 1;
                    sum = 0;
                }
            }
            len += 1;
        }
        return 0;       
    }
};
// @lc code=end

File: hengshun/209.长度最小的子数组.py
#
# @lc app=leetcode.cn id=209 lang=python3
#
# [209] 长度最小的子数组
#

# @lc code=start
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        i = j = 0
        res = n + 1
        subarr_sum = 0
        while j < n:
            subarr_sum += nums[j]
            j += 1
            while subarr_sum >= target:
                res = min(res, j - i)
                subarr_sum -= nums[i]
                i += 1
        if res == n + 1:
            return 0
        return res

# @lc code=end

